src/Services/AppInstallActivationHandler.cs

```diff
@@ -51,63 +63,107 @@ protected override bool CanHandleInternal(ProtocolActivatedEventArgs args)
+            _log.Warning("No valid identifiers provided in the query.");
             return;
         }
 
-        // Currently using the first search term only
-        var firstSearchTerm = searchParameter.Split(Separator, StringSplitOptions.RemoveEmptyEntries)
-                                             .Select(term => term.Trim(' ', '"'))
-                                             .FirstOrDefault();
+        _log.Information("Starting add-apps-to-cart activation");
+        _navigationService.NavigateTo(typeof(SetupFlowViewModel).FullName!);
+        _setupFlowViewModel.StartAppManagementFlow(queryType == ActivationQueryType.Search ? identifiers[0] : null);
+        await HandleAppSelectionAsync(identifiers, queryType);
+    }
 
-        if (string.IsNullOrEmpty(firstSearchTerm))
+    private async Task HandleAppSelectionAsync(string[] identifiers, ActivationQueryType queryType)
+    {
+        try
         {
-            _log.Warning("No valid search term was extracted from the query.");
-            return;
+            await SearchAndSelectAsync(identifiers);
         }
+        catch (Exception ex)
+        {
+            _log.Error(ex, "Error executing the add-apps-to-cart activation flow");
+        }
+    }
+
+    private string[] ParseIdentifiers(string query, ActivationQueryType queryType)
+    {
+        switch (queryType)
+        {
+            case ActivationQueryType.Search:
+                var terms = query.Split(Separator, StringSplitOptions.RemoveEmptyEntries);
+                if (terms.Length > 0)
+                {
+                    var firstTerm = terms[0].Replace("\"", string.Empty).Trim();
+                    return string.IsNullOrEmpty(firstTerm) ? Array.Empty<string>() : new[] { firstTerm };
+                }
+
+                return Array.Empty<string>();
+
+            case ActivationQueryType.WingetIds:
+                return query.Split(Separator, StringSplitOptions.RemoveEmptyEntries)
+                            .Select(id => id.Trim(' ', '"'))
+                            .ToArray();
 
-        var searchResults = await _windowsPackageManager.SearchAsync(firstSearchTerm, 1);
-        if (searchResults.Count == 0)
+            default:
+                _log.Warning("Unsupported activation query type: {QueryType}", queryType);
+                return Array.Empty<string>();
+        }
+    }
+
+    private async Task SearchAndSelectAsync(string[] identifiers)
+    {
+        if (identifiers == null || identifiers.Length == 0)
         {
-            _log.Warning("No results found for the search term: {SearchTerm}", firstSearchTerm);
+            _log.Warning("No valid identifiers provided in the query.");
             return;
         }
 
-        var firstResult = _packageProvider.CreateOrGet(searchResults[0]);
-        firstResult.IsSelected = true;
+        foreach (var identifier in identifiers)
+        {
+            var searchResults = await _windowsPackageManager.SearchAsync(identifier, 1);
```

Comment:
We can use `GetPackagesAsync` for more efficient search since we already know the Ids

Reply:
Done. Thanks for the pointers!